realm-transformer/src/main/kotlin/io/realm/transformer/RealmTransformer.kt

```diff
@@ -37,62 +57,99 @@ val READ_TIMEOUT = 2000L;
 // Wrapper for storing data from org.gradle.api.Project as we cannot store a class variable to it
 // as that conflict with the Configuration Cache.
 data class ProjectMetaData(
-    val isOffline: Boolean,
-    val bootClassPath: List<File>)
+        val bootClassPath: Set<File>,
+        val usesKotlin: Boolean,
+        val targetType: String,
+        val targetSdk: String,
+        val minSdk: String,
+        val agpVersion: String,
+        val appId: String,
+        val gradleVersion: String,
+        val usesSync: Boolean,
+        val isGradleOffline: Boolean) {
+}
 
 /**
  * This class implements the Transform API provided by the Android Gradle plugin.
  */
-class RealmTransformer(project: Project) : Transform() {
-    private val logger: Logger = LoggerFactory.getLogger("realm-logger")
-    private lateinit var metadata: ProjectMetaData
-    private var analytics: RealmAnalytics? = null
-
-    init {
-        // Fetch project metadata when registering the transformer, but as some of the properties
-        // we need to read might not be initialized yet (e.g. the Android extension), we need
-        // to wait until after the build files have been evaluated.
-        project.afterEvaluate {
-            metadata = ProjectMetaData(
-                // Plugin requirements
-                project.gradle.startParameter.isOffline,
-                project.getBootClasspath()
-            )
+class RealmTransformer(private val metadata: ProjectMetaData,
+                       private val inputs: ListProperty<Directory>,
+                       private val allJars: ListProperty<RegularFile>,
+                       private val referencedInputs: ConfigurableFileCollection,
+                       private val output: RegularFileProperty) {
 
-            try {
-                this.analytics = RealmAnalytics()
-                this.analytics!!.calculateAnalyticsData(project)
-            } catch (e: Exception) {
-                // Analytics should never crash the build.
-                logger.debug("Could not calculate Realm analytics data:\n$e" )
-            }
-        }
-    }
-    override fun getName(): String {
-        return "RealmTransformer"
+    private val analytics: RealmAnalytics? = try {
+        val analytics = RealmAnalytics()
+        analytics.calculateAnalyticsData(metadata)
+        analytics
+    } catch (e: Exception) {
+        // Analytics should never crash the build.
+        logger.debug("Could not calculate Realm analytics data:\n$e")
+        null
     }
 
-    override fun getInputTypes(): Set<QualifiedContent.ContentType> {
-        return setOf<QualifiedContent.ContentType>(QualifiedContent.DefaultContentType.CLASSES)
-    }
+    companion object {
 
-    override fun isIncremental(): Boolean {
-        return true
-    }
+        @Suppress("UnstableApiUsage")
```

Comment:
Nitpick ... but maybe we could remove that now. Maybe just for one of the follow up PRs bumping versions, etc.